### PR TITLE
Message list composer

### PR DESCRIPTION
**Pull request by LukasHromadnik**
# Description

* Added list update based on the changes from `ChatChannelController`
* Integrate message composer
* Implement scrolling to the latest message

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift [hunk 1]
```diff
@@ -24,37 +30,145 @@ class MessageListVC<ExtraData: ExtraDataTypes>: _ViewController, UICollectionVie
...
         let message = channelController.messages[indexPath.row]
         let isLastInGroup = isMessageLastInGroup(at: indexPath)
-
+        
         var options: ChatMessageLayoutOptions = []
-
+        
         if message.isSentByCurrentUser {
             options.insert(.flipped)
         }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift:174]: Strange the the formatter didn't remove these

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift [hunk 2]
```diff
@@ -118,4 +232,181 @@ class MessageListVC<ExtraData: ExtraDataTypes>: _ViewController, UICollectionVie
...
+                    collectionView.moveItem(at: fromIndex, to: toIndex)
+                case let .remove(_, index):
+                    collectionView.deleteItems(at: [index])
+                case let .update(_, index):
+                    collectionView.reloadItems(at: [index])
+                }
+            }
+        } completion: { flag in
+            completion?(flag)
+            self.scrollToMostRecentMessageIfNeeded()
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift:277]: This doesn't seem right. Why we should scroll to the last message when anything changes in the messages?

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift [hunk 3]
```diff
@@ -24,37 +30,145 @@ class MessageListVC<ExtraData: ExtraDataTypes>: _ViewController, UICollectionVie
...
+        view.backgroundColor = .white
         
-        view.embed(collectionView)
         collectionView.backgroundColor = .white
         
-        collectionView.register(MessageCell<ExtraData>.self, forCellWithReuseIdentifier: MessageCell<ExtraData>.reuseId)
+        navigationItem.titleView = titleView
+    }
+    
+    private func updateNavigationTitle() {
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift:105]: this needs to be called when something in the channel changes, too. You can just listen to `channel controller(didUpdateChannel:)` callbacks and call this.

**Review by VojtaStavik — APPROVED** [review]
All right 👍 Just some small fixes and you can merge this

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift (hunk 2 above)
Comment (LukasHromadnik) [reply Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift:277]: taken from current implementation in `ChatMessageListVC`

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift (hunk 3 above)
Comment (LukasHromadnik) [reply Sources/StreamChatUI/ChatMessageList/RETHINK/MessageListVC.swift:105]: ✅

File: Sources/StreamChat/Controllers/ChannelController/ChannelController.swift [hunk 4]
```diff
@@ -204,6 +204,10 @@ public class _ChatChannelController<ExtraData: ExtraDataTypes>: DataController,
             }
         }
     }
+    
+    public var isChannelDirect: Bool {
+        channel?.isDirectMessageChannel ?? false
+    }
```
[the remark this answers is not in the document]
Comment (LukasHromadnik) [reply Sources/StreamChat/Controllers/ChannelController/ChannelController.swift]: ✅

**Pull request merged**